### ISSUE TITLE
ChartData2D Component ImportFromSpreadsheet method — "useHeaders" parameter does not work

### ISSUE THREAD

**Issue by bobparks** [mit-cml/appinventor-sources#2775]
**Describe the bug**

If you set the userHeaders parameter as True (as shown in the attached AIA). Nothing happens. No data at all. Should it be reading the headers and setting them as labels for the axes? Or at least skip over them. 

If you set the parameter as False, and the spreadsheet has headers (as shown in the [referenced spreadsheet](https://docs.google.com/spreadsheets/d/1Hrxxa8kroOWKUQOA7x2OYJTtjxKSPweSWPqrKMahVL8/edit#gid=328833217)), then Chart throws up an error, as expected. You are trying to make it read strings as data. 

**Affects**

- [ ] Designer
- [ ] Blocks editor
- [x ] Companion
- [ ] Compiled apps
- [x] Buildserver
- [ ] Debugging
- [ ] Other... (please describe)

**Expected behavior**

Should it automatically label the axes in the Chart? Or should it at least skip over the headers? 

**Steps to reproduce**

Load and run the attached AIA. Press the Get Data button to load the data into your app. Then press Show Data button to load the data into the chart. 

[Spreadsheet_to_Chart_starter_header_test.zip](https://github.com/mit-cml/appinventor-sources/files/10059844/Spreadsheet_to_Chart_starter_header_test.zip)

**Comment by devarsh-mavani-19** [comment]
@bobparks this is very interesting issue.
I was debugging the code and I found that when the useHeaders is true, The Code is expecting the first value of columns in xColumn and yColumn so in your case the xColumn and yColumn should be "winter" and "ice_days" respectively (if useHeaders is true). But if you set useHeaders to false in that case you have to pass D and E (the actual spreadsheet column) as you have done in your aia. 
This makes it very inconsistent for students and new students will find it difficult. So we can do either of following
1. Update the documentation and specify that if useHeaders is true then xColumn and yColumn should be the frst values and not the column index (A-Z). While this is simple, It will make the overall experience in consistent and is not very intutive.
2. Second option is to make both the behaviour consistent (I would like to work on this issue if we choose to go with this approach). But the downside is that it will make existing projects in-consistent.

What are your thoughts on this @ewpatton sir?

**Comment by devarsh-mavani-19** [comment]
@ewpatton can you please take a look at above comment. Is this an expected behaviour or is it a bug?
I am referring to this part
> The Code is expecting the first value of columns in xColumn and yColumn so in your case the xColumn and yColumn should be "winter" and "ice_days" respectively (if useHeaders is true). But if you set useHeaders to false in that case you have to pass D and E (the actual spreadsheet column) as you have done in your aia.

Example
![image](https://user-images.githubusercontent.com/55136047/213452080-a165621a-4cfd-4d0e-8286-d2bf8dba11dc.png)
This is what it should be when useHeaders is false
![image](https://user-images.githubusercontent.com/55136047/213452227-5b9bf381-0cf2-4a0c-ae94-a55539ace5c9.png)
And this is when useHeaders is true
notice I changed the xColumn and yColumn name to actual columns inputted by user in spreadsheet instead of the default A-Z column names

**Comment by ewpatton** [comment]
@devarsh-mavani-19 Yes, that's correct. I felt like I had it working at some point but if you want to poke around and see if you can come up with a fix that would be helpful.

**Comment by bobparks** [comment]
Many thanks, @devarsh-mavani-19. I didn't realize the existing code was looking for "winter" or "ice_days". That's why it wasn't working for me. Great catch. 

Yes, I think making it consistent using an alphabetical column name is more intuitive.

**Comment by devarsh-mavani-19** [comment]
> Many thanks, @devarsh-mavani-19. I didn't realize the existing code was looking for "winter" or "ice_days". That's why it wasn't working for me. Great catch.
> 
> Yes, I think making it consistent using an alphabetical column name is more intuitive. And this change will make the block skip the header row when the useHeaders parameter is set to true?

Yes that's correct. But if we try to make behaviour consistent, It might cause issues in existing user projects.

**Comment by ewpatton** [comment]
@bobparks Let's plan to discuss this on Thursday. The semantics you're proposing sound more like they'd be in line with a `skipHeaders` parameter rather than a `useHeaders` parameter. My vision with `useHeaders` was to make it easier for people to reference data by the actual labels used in the data (assuming it's columnar) rather than have to fall back to the spreadsheet column lettering.
